frontend/javascripts/dashboard/dataset/dataset_import_view.js

```diff
@@ -94,6 +94,40 @@ export type FormData = {
   defaultConfigurationLayersJson: string,
 };
 
+function ensureValidScaleOnInferredDataSource(
+  savedDataSourceOnServer: APIDataSource | void,
+  inferredDataSource: APIDataSource | void,
+): APIDataSource | null {
+  if (savedDataSourceOnServer == null || inferredDataSource == null) {
+    return null;
+  }
+  const inferredDataSourceClone = (_.cloneDeep(inferredDataSource): any);
+  if (
+    _.isEqual(inferredDataSource.scale, [0, 0, 0]) &&
+    !_.isEqual(savedDataSourceOnServer.scale, [0, 0, 0])
+  ) {
+    inferredDataSourceClone.scale = savedDataSourceOnServer.scale;
+  }
+  // Trying to use the saved value for largestSegmentId instead of 0.
+  if (savedDataSourceOnServer.dataLayers != null && inferredDataSourceClone.dataLayers != null) {
+    const segmentationLayerSettings = inferredDataSourceClone.dataLayers.find(
+      layer => layer.category === "segmentation",
+    );
+    const savedSegmentationLayerSettings = savedDataSourceOnServer.dataLayers.find(
+      layer => layer.category === "segmentation",
+    );
+    if (
+      segmentationLayerSettings != null &&
+      savedSegmentationLayerSettings != null &&
+      segmentationLayerSettings.largestSegmentId === 0 &&
+      savedSegmentationLayerSettings.category === "segmentation"
```

Comment:
This additional check is needed because flow is not smart enough to know that savedSegmentationLayerSettings must be settings for a segmentation layer and not for a color layer.

I added a comment for this :+1: